### PR TITLE
arith: Proofs for Diophantine cuts

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Thread proofs through the diophantine "cutting" lemma generator.

### CODE REVIEW

File: src/theory/arith/theory_arith_private.h [hunk 1]
```diff
@@ -237,7 +237,7 @@ class TheoryArithPrivate {
   context::CDQueue<ArithVar> d_constantIntegerVariables;
 
   Node callDioSolver();
-  Node dioCutting();
+  TrustNode dioCutting();
```
Comment (ajreynol) [inline src/theory/arith/theory_arith_private.h:240]: Can you give some quick documentation on the form of this lemma?

**Review by ajreynol — CHANGES_REQUESTED** [review]
One documentation request.

File: src/theory/arith/theory_arith_private.h (hunk 1 above)
Comment (alex-ozdemir) [reply src/theory/arith/theory_arith_private.h:248]: Sure thing. Done in bf6184e25.

**Review by ajreynol — APPROVED** [review]

**Comment by ajreynol** [comment]
Thanks for the doc :)

**Pull request merged**